### PR TITLE
fix(plugin-manifest): Allow for all valid WebAppManifest properties

### LINKED ISSUE #27927: gatsby-plugin-manifest doesn't allow app shortcuts
## Description

Adding [app `shortcuts`](https://web.dev/app-shortcuts/) to the plugin options in gatsby-plugin-manifest 2.5.0+ results in an _Invalid plugin options_ error

I _think_ this was caused by enabling 'plugin option validation', released in #27437 and published in 2.5.0 of gatsby-plugin-manifest. The new options validation doesn't include the `shortcuts` option. It worked before 2.5.0 - I've just upgraded from 2.4.21 where it was working.

### Steps to reproduce

1. Install gatsby-plugin-manifest 2.5.0
2. Add `shortcuts` to the plugin's options:

```diff
// gatsby-config.s
module.exports = {
  plugins: [
    {
      resolve: `gatsby-plugin-manifest`,
      options: {
+        shortcuts: [
+          { name: "Topics A to Z", url: "/topics/?utm_source=a2hs&utm_medium=shortcuts", }
+        ]
      }
    }
  ]
}
```

### Expected result

Shortcuts are added to the generated _manifest.webmanifest_

### Actual result

You get the following CLI error:

> error Invalid plugin options for "gatsby-plugin-manifest": "shortcuts" is not allowed

### Environment

  System:
    OS: Windows 10 10.0.19041
    CPU: (4) x64 Intel(R) Core(TM) i7-5500U CPU @ 2.40GHz
  Binaries:
    Node: 12.16.3 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.16.0 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.14.4 - C:\Program Files\nodejs\npm.CMD
  Languages:
    Python: 2.7.11
  Browsers:
    Chrome: 86.0.4240.183
    Edge: Spartan (44.19041.423.0)
  npmPackages:
    gatsby: ^2.25.3 => 2.25.3 
    gatsby-plugin-catch-links: ^2.3.15 => 2.3.15 
    gatsby-plugin-eslint: ^2.0.8 => 2.0.8 
    gatsby-plugin-google-tagmanager: ^2.4.0 => 2.4.0 
    gatsby-plugin-manifest: ^2.5.2 => 2.5.2 
    gatsby-plugin-offline: ^3.3.2 => 3.3.2 
    gatsby-plugin-preact: ^4.0.16 => 4.0.16 
    gatsby-plugin-prefetch-google-fonts: ^1.4.3 => 1.4.3 
    gatsby-plugin-react-helmet: ^3.3.14 => 3.3.14 
    gatsby-plugin-sass: ^2.4.2 => 2.4.2 
    gatsby-plugin-sitemap: ^2.5.1 => 2.5.1 
    gatsby-plugin-split-css: ^1.0.2 => 1.0.2 
    gatsby-plugin-typescript: ^2.5.0 => 2.5.0 
    gatsby-react-router-scroll: 3.0.0 => 3.0.0

### PR DESCRIPTION
**Pull request by moonmeister**
## Description

Current Plugin Options Validator for `gatsby-plugin-manifest` is very incomplete. This PR implements a spec compliant validator.

### Documentation

I've added a not to docs about validation and the version of the spec it implements. 

##TODO:

This is copied form the previous PR...testing and updates are needed.

- [x] Update to the latest spec - https://www.w3.org/TR/2020/WD-appmanifest-20201019/
 - [ ] ~~Significant updates can now be made to how the plugin handles defaults, these can be done here or in a new PR.~~

- [ ] ~~add Docs about https://github.com/gatsbyjs/gatsby/pull/27885~~

## Related Issues

fixes #27927

### CODE REVIEW

**Comment by moonmeister** [comment]
This should be good. I'll handle changing how defaults are handled in another PR so we can get this fix out ASAP.

File: packages/gatsby-plugin-manifest/package.json
Comment (pvdz) [inline packages/gatsby-plugin-manifest/package.json:50]: prettier should fix this (you could tune your editor ;) )

**Review by pvdz — DISMISSED** [review]
nais :purple_heart:

File: packages/gatsby-plugin-manifest/package.json
Comment (pvdz) [reply packages/gatsby-plugin-manifest/package.json:50]: (won't show up in discussions but the line ending is missing here)

**Review by mxstbr — DISMISSED** [review]
SO GOOD!!! Thank you so much for digging into this :100:

**Comment by pvdz** [comment]
(ran formatter)

File: packages/gatsby-plugin-manifest/src/pluginOptionsSchema.js
Comment (ediblecode) [inline packages/gatsby-plugin-manifest/src/pluginOptionsSchema.js:30]: Nit pick: this is missing `.optional` which I know is the default but it's explicit everywhere else

File: packages/gatsby-plugin-manifest/src/pluginOptionsSchema.js
Comment (ediblecode) [inline packages/gatsby-plugin-manifest/src/pluginOptionsSchema.js:101]: The spec recommends

> Manifest authors are encouraged to use lower-case.

So could we consider using `.lowercase()` here?

File: packages/gatsby-plugin-manifest/src/pluginOptionsSchema.js
Comment (ediblecode) [inline packages/gatsby-plugin-manifest/src/pluginOptionsSchema.js:179]: There's a [supplementary spec](https://w3c.github.io/manifest-app-info/) (I know, not confusing at all) that contains extra members. The two relevant ones are platform and label. I suggest maybe we at least add label to screenshots because "For accessibility, authors are encouraged to provide a label for each screenshot."

File: packages/gatsby-plugin-manifest/src/pluginOptionsSchema.js
Comment (moonmeister) [reply packages/gatsby-plugin-manifest/src/pluginOptionsSchema.js:101]: I thought about this too. What I didn't want is it to throw an error if the user didn't put in all lowercase. But conversion should be enabled so it's probably fine.

Comment (ediblecode) [reply packages/gatsby-plugin-manifest/src/pluginOptionsSchema.js:101]: agreed, will resolve this comment

File: packages/gatsby-plugin-manifest/src/pluginOptionsSchema.js
Comment (moonmeister) [reply packages/gatsby-plugin-manifest/src/pluginOptionsSchema.js:179]: Good find. I'll get these added.

**Review by mxstbr — APPROVED** [review]
Looks fantastic to me!

**Comment by moonmeister** [comment]
This is pending a review from docs before the bot will merge.

**Comment by vladar** [comment]
Published in `gatsby-plugin-manifest@2.6.1`

**Pull request merged**
